### PR TITLE
Feature/maroon 233 fix interference plate when paused

### LINKED ISSUE #233: [HuygensPrinciple] Fix InterferenceDisplay When Paused
When the simulation is paused, the interference pattern is shown, and parameter changes are accepted.

### PR DESCRIPTION
**Pull request by dpdsw**
closes #233

### CODE REVIEW

File: unity/Assets/0Refactored/scenes/experiments/HuygensPrinciple/scripts/InterferencePlate.cs [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
 {
-    public class InterferencePlate : MonoBehaviour
+    public class InterferencePlate : PausableObject
     {
         [SerializeField]
         private SlitPlate slitPlate;
 
         private MeshRenderer _meshRenderer;
-
+        private bool _parameterHasChanged = false;
```
Comment (michaelholly) [inline unity/Assets/0Refactored/scenes/experiments/HuygensPrinciple/scripts/InterferencePlate.cs:11]: Initializing field by default is redundant.

File: unity/Assets/0Refactored/scenes/experiments/HuygensPrinciple/scripts/InterferencePlate.cs [hunk 2]
```diff
@@ -18,15 +19,27 @@ private void Start()
 
         private void Update()
```
Comment (michaelholly) [inline unity/Assets/0Refactored/scenes/experiments/HuygensPrinciple/scripts/InterferencePlate.cs:20]: Please remove the empty Update() function for performance reasons.

File: unity/Assets/0Refactored/scenes/experiments/HuygensPrinciple/scripts/InterferencePlate.cs [hunk 3]
```diff
@@ -18,15 +19,27 @@ private void Start()
...
         {
-            //ToDo: Add OnMovement Event to the SlitPlate script
+        }
+
+        protected override void HandleUpdate()
+        {
+        }
+
+        protected override void HandleFixedUpdate()
```
Comment (michaelholly) [inline unity/Assets/0Refactored/scenes/experiments/HuygensPrinciple/scripts/InterferencePlate.cs:28]: I think we can handle this in the HandleUpdate() method.
FixedUpdate is used for being in-step with the physics engine, so anything that needs to be applied to a rigidbody should happen in FixedUpdate. Update, on the other hand, works independently of the physics engine.

File: unity/Assets/0Refactored/scenes/experiments/HuygensPrinciple/scripts/InterferencePlate.cs [hunk 4]
```diff
@@ -18,15 +19,27 @@ private void Start()
...
+        {
             if (slitPlate.transform.hasChanged)
```
Comment (michaelholly) [inline unity/Assets/0Refactored/scenes/experiments/HuygensPrinciple/scripts/InterferencePlate.cs:30]: I think we can combine these two conditions.
e.g.: if(!slitPlate.transform.hasChanged && !_parameterHasChanged) return;

**Review by michaelholly — CHANGES_REQUESTED** [review]

**Comment by dpdsw** [comment]
Changes done. Also squashed the commits now.

**Review by michaelholly — APPROVED** [review]

**Pull request merged**